src/result.rs

```diff
@@ -1,94 +1,28 @@
 //! Error types that can be emitted from this library
 
-use std::convert;
-use std::error;
-use std::fmt;
 use std::io;
 
+use thiserror::Error;
+
 /// Generic result type with ZipError as its error variant
 pub type ZipResult<T> = Result<T, ZipError>;
 
 /// Error type for Zip
-#[derive(Debug)]
-pub enum ZipError
-{
+#[derive(Debug, Error)]
+pub enum ZipError {
     /// An Error caused by I/O
-    Io(io::Error),
+    #[error(transparent)]
+    Io(#[from] io::Error),
 
     /// This file is probably not a zip archive
+    #[error("Invalid Zip archive")]
```

Comment:
nit: it might be a better idea to have the display strings start with lowercase letters. This is what `thiserror` shows in their documentation, and I think it generally makes for slightly nicer error messages.

Reply:
ah, cheers for spotting that. I will fix it right away